### ISSUE TITLE
Match mzXML to MSRunSample using currently recorded metadata

### ISSUE THREAD

**Issue by hepcat72** [Princeton-LSI-ResearchComputing/tracebase#878]
### Inspiration

Lance and I talked about the last study load yesterday.  Of particular note was the fact that the file names had duplicates.  They were not distinguishable via suffix (`_pos`, `_neg`, `_scan#`, or high/low) and they were not organized in a way that associated them with the accucor/isocorr files they were used for.  I.e. There were files with identical names.

I was thinking about this last night, and I had a few thoughts:

- Each MSRunSample record is associated with:
  - polarity (which may be unknown, but can be derived)
  - peak data record with a med_mz
- mzXML files
  - The file name includes the sample name
  - The file contains polarity
  - The file contains a scan range
- accucor/isocorr files:
  - It has been empirically observed that each mzXML file traverses a scan range that is accurately (but imprecisely) equivalent for a set of samples used in a single accucor/isocorr analysis.
  - Contains an mz range in the medMz column (min and max of that column)
  - Contains sample names

### Description

I think it would be fairly straight-forward to match an mzXML file to its corresponding MSRunSample record.

Things to note:

1. We already require polarity when there are multiple scans of the same sample.  Currently, the researcher must specify that, which I think suffices, but we could probably automate that deduction if we wanted to.
2. Right now, the LCMS metadata (which includes associating mzXML files with samples) is processed before the accucor data is traversed & loaded and before the MSRunSample records are created, but the dataframes are provided at the start of the loading code, which makes all the data available.  Those dataframes are already pre-processed to extract the sample headers, and we could make it also extract the medMz range.
3. And there already exists code that parses polarity and scan range from the mzXML files.

All we would have to do is:

1. Add the ability to extract the medMz range from the accucor/isocorr file
2. As we come to each mzXML file, parse it for its polarity, mz_min, and mz_max
3. If the polarity matched and the median medMz from the accucor/isocorr file match, and the file name matches one of the sample names pre-extracted from the accucor/isocorr file as well, associate the mzXML with that header.  If it doesn't, it's an extra, and can be skipped (like it currently does for extra files)^

^ Extra mzXML files are accounted for when the sample sheet is processed

### Alternatives

None

### Dependencies

This issue cannot be started until the completion of the following issue(s)/
pull request(s):

- #issue_number_1
- #pull_request_1

### Comment

None

-----

### Assumptions

1. List of assumptions that the code will not explicitly address/check
4. E.g. We will assume input is correct (explaining why there is no validation)

### Limitations

1. A list of things this work will specifically not do
4. E.g. This feature will only handle the most frequent use case X

### Affected Components

- change: File path or DB table ...
- add: Environment variable or server setting
- delete: External executable or cron job

### Requirements

- [ ] 1. List of numbered conditions to be met for the feature
- [ ] 2. E.g. Every column/row must display a value, i.e. cannot be empty
- [ ] 3. Numbers for reference & checkboxes for progress tracking

#### Interface Change description

None provided

#### Code Change Description

None provided

#### Tests

- [ ] 1. A description of at least one test for each requirement above.
- [ ] 2. E.g. Test for req 2 that there's an exception when display value is ''
- [ ] 3. Numbers for reference & checkboxes for progress tracking

**Comment by hepcat72** [comment]
Even if we had all the info, there's no point in trying to identify the correct record to link peak groups to.  2 runs on the same day would be indistinguishable, and scan ranges can overlap.  We decided recently that given all that, we would always just link peak groups to a placeholder and we would re-implement the "bucket of [mzXML] files".

**Comment by jcmatese** [comment]
Sorry, just noticed this issue was closed, but we can talk about it if it ever comes up again in future design...

**Comment by hepcat72** [comment]
No problem, but let me elaborate. When I discussed this with Lance maybe a month ago or so, Lance was pointing out that there's just no way to discreetly and uniquely collate metadata to be able to associate peak groups with the MS run Sample record associated with the MZXML file from which peak data was derived. If the median MZ of the data in the Peak annotation file falls between the scan ranges of 2 MZXL files whose scan rages overlap, then there's no way to select one over the other. That is regardless of the fact that we still cannot distinguish two runs from the same day with the same Protocol. So what this issue is describing is simply not fully possible.

That said, my usage of "placeholder" is somewhat of a misnomer in this context. If we re-implement the bucket of files, all there are, are "placeholders". The design goes back to essentially what we had already before we changed the many too many relationship between MS run sample records and archive file records. For these reasons, I closed this issue as not planned.

(I was going through issues yesterday to assign  Component labels and to make sure that milestones were appropriately assigned. I closed a few issues that were either outdated or redundant, this one among them.)
